### ISSUE TITLE
[clang] Add a diagnostic flag to ensure "publically" accessible symbols are documented

### ISSUE THREAD

**Issue by NekkoDroid** [llvm/llvm-project#62968]
Currently there is the [`-Wdocumentation` flag](https://clang.llvm.org/docs/DiagnosticsReference.html#wdocumentation) that ensures documented symbols are valid doc comments. But there is currently no way to get warned if any accssible symbols are undocumented (other than running doxygen).

What exactly I mean with accessible symbols are:
- Types and their public and protected members
- Non-static functions defined in any non-anonymous namespace

Basically anything, that is accessible from one TU in a different TU.
(This may be also limited to symbols declared in headers since else they would need to be redeclared anyway)
